recipes/physx/4.1.1/conanfile.py

```diff
@@ -0,0 +1,304 @@
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        del self.options.fPIC # fpic is managed by physx build process (in physx/source/compiler/cmake/CMakeLists.txt)
+        if self.settings.build_type != "Release":
+            del self.options.release_build_type
+        if self.settings.os != "Windows":
+            del self.options.enable_float_point_precise_math
+        if self.settings.os not in ["Windows", "Android"]:
+            del self.options.enable_simd
+
+    def configure(self):
+        os = str(self.settings.os)
+        if os not in ["Windows", "Linux", "Macos", "Android", "iOS"]:
+            raise ConanInvalidConfiguration("{0} {1} is not supported on {2}".format(self.name, self.version, os))
+
+        build_type = str(self.settings.build_type)
+        if build_type not in ["Debug", "RelWithDebInfo", "Release"]:
+            raise ConanInvalidConfiguration("{0} {1} does not support {2} build type".format(self.name, self.version,
+                                                                                             build_type))
+
+        compiler = str(self.settings.compiler)
+        if os == "Windows" and compiler != "Visual Studio":
+            raise ConanInvalidConfiguration("{0} {1} does not support {2} on {3}".format(self.name, self.version,
+                                                                                         compiler, os))
+
+        if compiler == "Visual Studio":
+            if tools.Version(self.settings.compiler.version) < 9:
+                raise ConanInvalidConfiguration("{0} {1} does not support Visual Studio < 9".format(self.name,
+                                                                                                    self.version))
+
+            runtime = str(self.settings.compiler.runtime)
+            if build_type == "Debug":
+                if runtime not in ["MDd", "MTd"]:
+                    raise ConanInvalidConfiguration("Visual Studio Compiler runtime MDd or MTd " \
+                                                    "is required for {0} build type".format(build_type))
+            elif runtime not in ["MD", "MT"]:
+                raise ConanInvalidConfiguration("Visual Studio Compiler runtime MD or MT " \
+                                                "is required for {0} build type".format(build_type))
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        url = self.conan_data["sources"][self.version]["url"]
+        extracted_dir = "PhysX-" + os.path.splitext(os.path.basename(url))[0]
+        try:
+            os.rename(extracted_dir, self._source_subfolder)
+        except:
+            # workaround for permission denied on windows
+            time.sleep(10)
+            os.rename(extracted_dir, self._source_subfolder)
+        for patch in self.conan_data["patches"][self.version]:
```

Comment:
Apply your patches on line 124, before to configure. Keep sources immutable as possible,

Reply:
done